lib/compile/formats.js

```diff
@@ -33,7 +33,8 @@ formats.fast = {
   // email (sources from jsen validator):
   // http://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address#answer-8829363
   // http://www.w3.org/TR/html5/forms.html#valid-e-mail-address (search for 'willful violation')
```

Comment:
I think both are valid, strictly speaking... You won't be able to use my_domain.com as a host-name (as a part of URL) but you should be able to use it as a domain name. But practically given that everybody uses domains as hostnames maybe we shouldn't allow underscores in the second level and only allow it in the third etc.